src/analysis/special_functions/log/basic.lean

```diff
@@ -205,6 +205,25 @@ begin
   rw exp_log hx,
 end
 
+lemma log_nat_eq_factorization_sum {n : ℕ} : log n = n.factorization.sum (λ p t, t * log p) :=
+begin
+  refine n.rec_on_prime_pow _ _ (λ a p n hp hpa hn ih, _),
+  { simp },
+  { simp },
+  push_cast,
+  have ha : a ≠ 0,
+  { rintro rfl, exact hpa (dvd_zero p) },
+  have hpn : p ^ n ≠ 0 := pow_ne_zero n hp.ne_zero,
+  rw [log_mul, log_pow, nat.factorization_mul hpn ha, finsupp.sum_add_index', hp.factorization_pow,
+      finsupp.sum_single_index, ih],
+  { simp },
+  { simp },
+  { intros t x y,
+    rw [x.cast_add y, add_mul] },
+  { exact_mod_cast hpn },
+  { exact_mod_cast ha },
+end
```

Comment:
```suggestion
begin
  refine n.rec_on_prime_coprime _ (λ p n pp, _) (λ a b ha hb hab Ha Hb, _) ,
  { simp },
  { simp [pp.factorization_pow] },
  { push_cast,
    rw [log_mul, Ha, Hb, nat.factorization_mul_of_coprime hab],
    rw sum_add_index_of_disjoint _ _ _ (nat.factorization_disjoint_of_coprime hab),
    { simp [ne_zero_of_lt ha] },
    { simp [ne_zero_of_lt hb] } },
end
```

Reply:
`sum_add_index_of_disjoint` is added in #14786, which I've just PRed

Reply:
awesome, I'll wait for that then

Reply:
never mind, Bolton's suggestion seems a bit better! but definitely good to have the lemma too